lr-cli/cortx_setup/commands/node/finalize.py

```diff
@@ -35,86 +35,189 @@ class NodeFinalize(Command):
+        for user in ['nodeadmin', 'support']:
+            home_dir = Path(f'/home/{user}')
+#            ssh_dir = home_dir / '.ssh'
+
+            ugroup = user if user == "nodeadmin" else "root"
+
+            self.logger.debug(
+               f"Creating, if not present, a new user group '{ugroup}'"
+            )
+            StateFunExecuter.execute(
+                'group.present',
+                fun_kwargs=dict(
+                    name=ugroup
+                ),
+                targets=node_id
+            )
+
+            self.logger.debug(f"Creating sudoers file with permissions for user: '{user}'")
+            StateFunExecuter.execute(
+                'file.managed',
+                fun_kwargs=dict(
+                    name=f"/etc/sudoers.d/{user}",
+                    contents=(f'## Restricted access for {user} group users \n'
+                              f'%{user}   ALL = NOPASSWD: '
+                              '/opt/seagate/cortx/provisioner/node_cli/'),
+                    create=True,
+                    replace=True,
+                    user=ugroup,
+                    group=ugroup,
+                    mode=440
+                ),
+                targets=node_id
+            )
+
+            self.logger.debug(f"Creating user: '{user}'")
+            StateFunExecuter.execute(
+                'user.present',
+                fun_kwargs=dict(
+                    name=user,
+                    password=user_passwd,
+                    hash_password=True,
+                    home=str(home_dir),
+                    groups=ugroup,
+                    shell="/opt/seagate/cortx/provisioner/node_cli/"
+                ),
+                targets=node_id,
+                secure=True
+            )
+
+            self.logger.debug(f"Success: User creation for '{user}'")
+
+
     def run(self, force=False):
-        local_minion = local_minion_id()
-        self._validate_cert_installation()
-        self._validate_health_map()
-        self._validate_interfaces(local_minion)
-        s_type = self._validate_server_type(local_minion)
-        self._validate_devices(local_minion, s_type)
-        self.logger.debug("Done")
+        node_id = local_minion_id()
+        val_1 = self._validate_cert_installation()
+        val_2 = self._validate_health_map()
+        val_3 = self._validate_interfaces(node_id)
+        s_type = self._validate_server_type(node_id)
+        val_4 = self._validate_devices(node_id, s_type)
+
+        if (val_1 and val_2 and val_3 and val_4):
+            self.logger.info(
+                "Node validations complete. Proceeding to create users.."
+            )
+        else:
+            if force:
+                self.logger.info(
+                    "One or more node validation(s) failed. Forcibly creating users.."
+                )
+            else:
+                raise CortxSetupError(
+                    "One or more node validation(s) failed. Check logs for more details."
```

Comment:
We should raise actual Exception exactly what failed?

Reply:
Addressed